### PR TITLE
Implement Column resource for SGv2/REST

### PR DESCRIPTION
**Pull request by tatu-at-datastax**
**What this PR does**:

Implements Columns CRUD endpoint for SGv2 REST (v2)

**Which issue(s) this PR fixes**:

**Checklist**
- [ ] Changes manually tested
- [x] Automated Tests added/updated -- another 10 or so ITs now pass for Column CRUD operations
- [ ] Documentation added/updated

### CODE REVIEW

File: sgv2-restapi/src/main/java/io/stargate/sgv2/restsvc/resources/ResourceBase.java [hunk 1]
```diff
@@ -34,6 +37,46 @@
...
+            final String msg = e.getMessage();
+            if (msg.contains("Keyspace not found")) {
+              return jaxrsBadRequestError(String.format("Keyspace '%s' not found", keyspaceName))
+                  .build();
+            }
+            return jaxrsBadRequestError(
+                    String.format(
+                        "Table: '%s' not found (in keyspace %s)", tableName, keyspaceName))
+                .build();
+        }
```
Comment (ivansenic) [inline sgv2-restapi/src/main/java/io/stargate/sgv2/restsvc/resources/ResourceBase.java:68]: would this expose a possibility to actually get the understanding if there is an existing keyspace/table information without actually being authorized to do so?

File: sgv2-restapi/src/main/java/io/stargate/sgv2/restsvc/resources/Sgv2RequestHandler.java [hunk 2]
```diff
@@ -40,21 +40,27 @@
...
       return action.call();
-    } catch (StatusRuntimeException grpcE) { // from gRPC
-      return failResponseForGrpcException(grpcE);
-    } catch (NotFoundException nfe) { // does this actually occur?
+    } catch (Exception e) {
+      return failResponseForAnyException(e);
+    }
+  }
+
+  public static Response failResponseForAnyException(Exception e) {
```
Comment (ivansenic) [inline sgv2-restapi/src/main/java/io/stargate/sgv2/restsvc/resources/Sgv2RequestHandler.java:48]: would this me not needed once the #1474 is merged?

File: sgv2-restapi/src/main/java/io/stargate/sgv2/restsvc/resources/schemas/Sgv2ColumnsResourceImpl.java [hunk 3]
```diff
@@ -0,0 +1,282 @@
...
+@Path("/v2/schemas/keyspaces/{keyspaceName}/tables/{tableName}/columns")
+@Produces(MediaType.APPLICATION_JSON)
+@Singleton
+@CreateGrpcStub
+public class Sgv2ColumnsResourceImpl extends ResourceBase implements Sgv2ColumnsResourceApi {
+  // Singleton resource so no need to be static
+  protected final Logger logger = LoggerFactory.getLogger(getClass());
+
+  @Override
+  public Response getAllColumns(
```
Comment (ivansenic) [inline sgv2-restapi/src/main/java/io/stargate/sgv2/restsvc/resources/schemas/Sgv2ColumnsResourceImpl.java:35]: OK I might be missing something around the authorization I guess.. where is it done?

File: sgv2-restapi/src/main/java/io/stargate/sgv2/restsvc/resources/schemas/Sgv2ColumnsResourceImpl.java [hunk 4]
```diff
@@ -0,0 +1,282 @@
...
+      return jaxrsBadRequestError("columnName must be provided").build();
+    }
+    return callWithTable(
+        blockingStub,
+        keyspaceName,
+        tableName,
+        (tableDef) -> {
+          Sgv2ColumnDefinition column = findColumn(tableDef, columnName);
+          if (column == null) {
+            return jaxrsBadRequestError(
```
Comment (ivansenic) [inline sgv2-restapi/src/main/java/io/stargate/sgv2/restsvc/resources/schemas/Sgv2ColumnsResourceImpl.java:130]: was this already 400? Always kind of confused in SG if such things should be `404`

File: sgv2-restapi/src/main/java/io/stargate/sgv2/restsvc/resources/schemas/Sgv2ColumnsResourceImpl.java [hunk 5]
```diff
@@ -0,0 +1,282 @@
...
+    }
+    if (isStringEmpty(columnName)) {
+      return jaxrsBadRequestError("columnName must be provided").build();
+    }
+    return callWithTable(
+        blockingStub,
+        keyspaceName,
+        tableName,
+        (tableDef) -> {
+          final String newName = columnUpdate.getName();
```
Comment (ivansenic) [inline sgv2-restapi/src/main/java/io/stargate/sgv2/restsvc/resources/schemas/Sgv2ColumnsResourceImpl.java:161]: short-circuit if new name is the same as old?

**Review by ivansenic — COMMENTED** [review]
Looks good, I am a bit confused around auth..

File: sgv2-restapi/src/main/java/io/stargate/sgv2/restsvc/resources/ResourceBase.java (hunk 1 above)
Comment (tatu-at-datastax) [reply sgv2-restapi/src/main/java/io/stargate/sgv2/restsvc/resources/ResourceBase.java:68]: There is a bigger discussion to be had about access, yes. But on short term gRPC API does check access rights AFAIK. There will be challenges in caching wrt tenancy.

Handling here is meant to replace SGv1 checks and produce same HTTP response codes, FWTW.
And try to extract it into shared handling, away from endpoint implementation.

File: sgv2-restapi/src/main/java/io/stargate/sgv2/restsvc/resources/Sgv2RequestHandler.java (hunk 2 above)
Comment (tatu-at-datastax) [reply sgv2-restapi/src/main/java/io/stargate/sgv2/restsvc/resources/Sgv2RequestHandler.java:48]: Yes, possibly; I haven't looked into that yet. I think there are trade-offs between local and global handling but I think you are right, for "anything else" cases global handler is probably better.

File: sgv2-restapi/src/main/java/io/stargate/sgv2/restsvc/resources/schemas/Sgv2ColumnsResourceImpl.java (hunk 3 above)
Comment (tatu-at-datastax) [reply sgv2-restapi/src/main/java/io/stargate/sgv2/restsvc/resources/schemas/Sgv2ColumnsResourceImpl.java:35]: by gRPC endpoint that is used for all Persistence access.

File: sgv2-restapi/src/main/java/io/stargate/sgv2/restsvc/resources/schemas/Sgv2ColumnsResourceImpl.java (hunk 4 above)
Comment (tatu-at-datastax) [reply sgv2-restapi/src/main/java/io/stargate/sgv2/restsvc/resources/schemas/Sgv2ColumnsResourceImpl.java:130]: Yes, I am following what Integration tests check. I agree it's not necessarily consistent or maybe even well-documented. I think that 404 is to be returned when directly trying to access non-existing thing (getKeyspace("nosuchks")), whereas 400 (bad request) would be when it's secondary target (update table where keyspace name is invalid). But ... yeah that's not a great explanation.

File: sgv2-restapi/src/main/java/io/stargate/sgv2/restsvc/resources/schemas/Sgv2ColumnsResourceImpl.java (hunk 5 above)
Comment (tatu-at-datastax) [reply sgv2-restapi/src/main/java/io/stargate/sgv2/restsvc/resources/schemas/Sgv2ColumnsResourceImpl.java:161]: Will do, sure.

File: sgv2-restapi/src/main/java/io/stargate/sgv2/restsvc/resources/Sgv2RequestHandler.java (hunk 2 above)
Comment (tatu-at-datastax) [reply sgv2-restapi/src/main/java/io/stargate/sgv2/restsvc/resources/Sgv2RequestHandler.java:48]: Yes, there's a non trivial conflict; need to resolve post-merge.

**Pull request merged**